### PR TITLE
[TF] Fix MaskRCNN QAT

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
### Changes

Fixed MaskRCNN QAT failing on Tensorflow. This was a regression after #1048.

### Related tickets

108752

### Tests

Changed MaskRCNN TF graph tests to operate on the model in training mode. This way we will catch bugs in QAT. Inference bugs should also be caught, but otherwise weekly will fail nevertheless. This leads to a slight improvement of TF precommit runtime: 54min. -> 57min.

### CODE REVIEW

File: nncf/tensorflow/graph/converter.py [hunk 1]
```diff
@@ -760,11 +761,11 @@ def _get_conv_layer_attributes(layer: tf.keras.layers.Layer, is_depthwise: bool
                                       padding_values=([0, 0, 0, 0]))
 
 
-def _get_linear_layer_attributes(layer: tf.keras.layers.Layer) -> LinearLayerAttributes:
+def _get_linear_layer_attributes(layer: tf.keras.layers.Layer, layer_metatype) -> LinearLayerAttributes:
     channel_axis = get_input_channel_axis(layer)
     in_features = layer.get_input_shape_at(0)[channel_axis]
     out_features = layer.get_output_shape_at(0)[channel_axis]
-    bias = layer.use_bias
+    bias = False if layer_metatype == op_metatypes.TFMatMulOpMetatype else layer.use_bias
```
Comment (nikita-savelyevv) [inline nncf/tensorflow/graph/converter.py:768]: For TFOpLambda layer, there's no `use_bias` property. Added a workaround. This is a regression after #1319.

File: nncf/tensorflow/graph/converter.py [hunk 2]
```diff
@@ -760,11 +761,11 @@ def _get_conv_layer_attributes(layer: tf.keras.layers.Layer, is_depthwise: bool
                                       padding_values=([0, 0, 0, 0]))
 
 
-def _get_linear_layer_attributes(layer: tf.keras.layers.Layer) -> LinearLayerAttributes:
+def _get_linear_layer_attributes(layer: tf.keras.layers.Layer, layer_metatype) -> LinearLayerAttributes:
```
Comment (vshampor) [inline nncf/tensorflow/graph/converter.py:764]: Type hint missing

**Review by vshampor — CHANGES_REQUESTED** [review]

File: nncf/tensorflow/graph/converter.py (hunk 2 above)
Comment (nikita-savelyevv) [reply nncf/tensorflow/graph/converter.py]: Done

**Review by vshampor — APPROVED** [review]

File: nncf/tensorflow/graph/metatypes/common.py [hunk 3]
```diff
@@ -79,9 +78,7 @@
 ]
 
 LINEAR_LAYER_METATYPES = [
-    layer_metatypes.TFDenseLayerMetatype,
-    # LINEAR_TF_OP_METATYPES
-    op_metatypes.TFMatMulOpMetatype,
+    layer_metatypes.TFDenseLayerMetatype
```
Comment (alexsu52) [inline nncf/tensorflow/graph/metatypes/common.py:81]: Could you clarify if op_metatypes.TFMatMulOpMetatype operations will be quantized?

**Review by alexsu52 — CHANGES_REQUESTED** [review]
Let's discuss offline this PR.

File: nncf/tensorflow/graph/converter.py [hunk 4]
```diff
@@ -760,11 +761,12 @@ def _get_conv_layer_attributes(layer: tf.keras.layers.Layer, is_depthwise: bool
...
 
 
-def _get_linear_layer_attributes(layer: tf.keras.layers.Layer) -> LinearLayerAttributes:
+def _get_linear_layer_attributes(layer: tf.keras.layers.Layer, layer_metatype: Type[OperatorMetatype]) -> \
+        LinearLayerAttributes:
     channel_axis = get_input_channel_axis(layer)
     in_features = layer.get_input_shape_at(0)[channel_axis]
     out_features = layer.get_output_shape_at(0)[channel_axis]
-    bias = layer.use_bias
+    bias = False if layer_metatype is op_metatypes.TFMatMulOpMetatype else layer.use_bias
```
[the remark this answers is not in the document]
Comment (nikita-savelyevv) [reply nncf/tensorflow/graph/converter.py]: Done
